### PR TITLE
Add Cloudflare cards to Home, Stats, and Marketing/Traffic

### PR DESCRIPTION
**Pull request by donlair**
This PR implements @sfougnier's final Cloudflare card layouts details in pbAPfg-1fn-p2.

The changes include:
* New task illustration for the card on My Home.
* Upgrade nudges added to the Cloudflare cards on the Settings and Marketing pages.
* Final destination URLs for the learn more links.

Screenshots:

The Cloudflare task card on My Home:
Screen Shot 2021-02-16 at 12 54 00 PM

Learn more banner on Stats:
Screen Shot 2021-02-16 at 12 58 51 PM

Upgrade banner on Stats:
Screen Shot 2021-02-16 at 12 59 03 PM

Marketing > Traffic page:
Screen Shot 2021-02-16 at 12 59 45 PM

Settings > Performance (upsell view):
Screen Shot 2021-02-16 at 1 15 52 PM

Settings > Performance (Learn more view):
Screen Shot 2021-02-16 at 1 23 31 PM

#### Testing instructions

* Patch your sandbox with 55161-code.
* Sandbox public-api.
* Checkout this PR.
* Force Cloudflare task card to show on My Home via `/home/SITESLUG?dev=true&view=VIEW_CLOUDFLARE`
* Navigate to `/stats/day/SITESLUG` and with Premium and Free plan accts and confirm that the appropriate banner shows (upsell vs learn more)

### CODE REVIEW

File: client/my-sites/site-settings/cloudflare.js [hunk 1]
```diff
@@ -30,32 +40,74 @@ const Cloudflare = () => {
...
-					<Card>
+					{ ! sitePlan && <QuerySitePlans siteId={ siteId } /> }
+					<SettingsSectionHeader title={ translate( 'CDN' ) } />
+					<CompactCard>
 						<div className="site-settings__cloudflare">
+							<div className="site-settings__cloudflare-illustration">
+								<img src={ jetpackIllustration } alt="" />
+							</div>
 							<div className="site-settings__cloudflare-text">
+								<p className="site-settings__cloudflare-title">Jetpack Site Accelerator</p>
```
Comment (sixhours) [inline client/my-sites/site-settings/cloudflare.js]: Do we need to make this string available for translation even though it's a product name?

File: client/my-sites/site-settings/cloudflare.js [hunk 2]
```diff
@@ -30,32 +40,74 @@ const Cloudflare = () => {
...
+							forceDisplay={ true }
+						/>
+					) }
+					<CompactCard>
+						<div className="site-settings__cloudflare">
+							<div className="site-settings__cloudflare-illustration">
+								<img src={ cloudflareIllustration } alt="" />
+							</div>
+							<div className="site-settings__cloudflare-text">
+								<p className="site-settings__cloudflare-title">Cloudflare CDN</p>
```
Comment (sixhours) [inline client/my-sites/site-settings/cloudflare.js]: Same question about i18n for proper names...

File: client/my-sites/stats/cloudflare.js [hunk 3]
```diff
@@ -2,52 +2,62 @@
...
+							description={ translate(
+								'Cloudflare Analytics give you deeper insights into your site traffic and performance.'
+							) }
+							showIcon={ true }
+							disableCircle={ true }
+							iconPath={ cloudflareIllustration }
+							href="CLOUDFLARELEARNMORELINK"
+							tracksImpressionName="banner_stats_cloudflare"
+							event="calypso_stats_cloudflare_analytics_learn_more"
+							callToAction="Learn more"
```
Comment (sixhours) [inline client/my-sites/stats/cloudflare.js]: This should be wrapped in the translate function.

File: client/my-sites/stats/cloudflare.js [hunk 4]
```diff
@@ -2,52 +2,62 @@
...
+						<Banner
+							title={ translate( 'Discover more stats with Cloudflare Analytics' ) }
+							description={ translate(
+								'Cloudflare Analytics give you deeper insights into your site traffic and performance.'
+							) }
+							showIcon={ true }
+							disableCircle={ true }
+							iconPath={ cloudflareIllustration }
+							href="CLOUDFLARELEARNMORELINK"
+							tracksImpressionName="banner_stats_cloudflare"
```
Comment (sixhours) [inline client/my-sites/stats/cloudflare.js]: Does this need to follow the Tracks naming conventions? ie. `calypso_stats_cloudflare_banner_view` or something like that.

File: client/my-sites/stats/cloudflare.js [hunk 5]
```diff
@@ -2,52 +2,62 @@
...
+							event="calypso_stats_cloudflare_analytics_learn_more"
```
Comment (sixhours) [inline client/my-sites/stats/cloudflare.js]: For the Tracks naming conventions I think you need to include the action `_click` at the end.

File: client/my-sites/stats/cloudflare.js [hunk 6]
```diff
@@ -2,52 +2,62 @@
...
-						<Button onClick={ recordClick } href="CLOUDFLARELINK" target="_blank">
-							{ translate( 'Learn More' ) }
-						</Button>
-					</div>
-					{ isDesktop() && (
-						<div className="stats__card-illustration">
-							<img src={ cloudflareIllustration } alt="" />
-						</div>
+							customerType="business"
+							tracksImpressionName="upsell_nudge_stats_cloudflare"
```
Comment (sixhours) [inline client/my-sites/stats/cloudflare.js]: Same notes about naming conventions for the impression and click events here.

File: client/my-sites/site-settings/cloudflare.js [hunk 7]
```diff
@@ -30,32 +40,74 @@ const Cloudflare = () => {
...
+					{ sitePlan && showUpsell && (
+						<UpsellNudge
+							title={ translate( 'Available with Premium plans or higher' ) }
+							description={ translate(
+								'A CDN (Content Delivery Network) optimizes your content to provide users with the fastest experience.'
+							) }
+							href={ upgradeLink }
+							event={ 'calypso_settings_cloudflare_cdn_upsell_nudge' }
+							showIcon={ true }
+							forceDisplay={ true }
```
Comment (sixhours) [inline client/my-sites/site-settings/cloudflare.js]: I'm not sure you need to set `forceDisplay` here...IIRC it was only needed for some cases in the sidebar upsell nudges.

File: client/my-sites/site-settings/cloudflare.js [hunk 8]
```diff
@@ -30,32 +40,74 @@ const Cloudflare = () => {
...
+							</div>
+						</div>
+					</CompactCard>
+					{ sitePlan && showBizUpsell && (
+						<UpsellNudge
+							title={ translate( 'Available on Business plan or higher' ) }
+							href={ upgradeLink }
+							event={ 'calypso_settings_cloudflare_cdn_upsell_nudge' }
+							showIcon={ true }
+							forceDisplay={ true }
```
Comment (sixhours) [inline client/my-sites/site-settings/cloudflare.js]: I'm not sure you need to set `forceDisplay` here...IIRC it was only needed for some cases in the sidebar upsell nudges. I could be wrong, tho. :)

**Comment by sixhours** [comment]
This works as expected! I was wondering if we should include an icon on the Discover banner just for consistency; otherwise there's extra whitespace:

Screen Shot 2021-02-16 at 4 19 55 PM

The whitespace before the copy on this card also looks off to me: 



Maybe making the image a bit smaller and aligning the text to the top would help, but /cc @sfougnier in case you had thoughts!

I also noticed the links to learn more from the Cloudflare banners don't work; I'm assuming those are placeholders? Might be a good idea to note that in the code as a comment or an `@TODO` so we don't forget and accidentally launch a link to `cloudflarelink.com` :D

File: client/my-sites/stats/cloudflare.js [hunk 9]
```diff
@@ -2,52 +2,62 @@
...
+							event="calypso_stats_cloudflare_analytics_upsell_nudge"
+							showIcon={ true }
+							forceDisplay={ true }
+							callToAction="Upgrade"
```
Comment (sixhours) [inline client/my-sites/stats/cloudflare.js]: This string needs to be wrapped in a translate function.

**Comment by ghost** [comment]
Thanks for the ping @sixhours ! 

image

This banner appears to be using upsell styling, yet this isn't an upsell. It also should be dismissable. Can we add the Cloudflare symbol here and make this dismissable to reflect the design below?

Screen Shot 2021-02-16 at 5 08 51 PM

**Cloudflare settings card**
There is a lot of vertical padding here that seems to be created by the size of the logo.

- We should scale that logo down significantly, as reflected in the design.
- Using the left-aligned logo layout in the design may help alleviate any strange white space or eye movement created by placing it on the right.

We should either remove that padding above the paragraph or add back in the header style.
Screen Shot 2021-02-16 at 5 08 01 PM

**Comment by blackjackkent** [comment]
@donlair This looks fine to me and tests correctly as advertised but we're going to want another pass on this to address the designers' comments before we merge.

> Maybe making the image a bit smaller and aligning the text to the top would help, but /cc @sfougnier in case you had thoughts!

@sixhours This design for the second image you linked is updated as part of https://github.com/Automattic/wp-calypso/pull/49629 and not part of this PR's testing instructions.

> I also noticed the links to learn more from the Cloudflare banners don't work; I'm assuming those are placeholders? Might be a good idea to note that in the code as a comment or an @TODO so we don't forget and accidentally launch a link to cloudflarelink.com :D

Yep. I'm gonna create a ticket for this purpose right now.

**Comment by donlair** [comment]
@sixhours @blackjackkent I just committed changes that I believe address all of Caroline's feedback.

Looking at @sfougnier's feedback now.

File: client/my-sites/site-settings/cloudflare.js [hunk 10]
```diff
@@ -30,32 +40,76 @@ const Cloudflare = () => {
...
+						/>
+					) }
+					<CompactCard>
+						<div className="site-settings__cloudflare">
+							<div className="site-settings__cloudflare-illustration">
+								<img src={ cloudflareIllustration } alt="" />
+							</div>
+							<div className="site-settings__cloudflare-text">
+								<p className="site-settings__cloudflare-title">
+									{ translate( 'Cloudflare CDN ' ) }
```
Comment (a8ci18n) [inline client/my-sites/site-settings/cloudflare.js:83]: :information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **18** times:
`translate( 'Cloudflare CDN' )` *ES Score: 13*

**Comment by donlair** [comment]
Thanks for the reviews @sixhours and @sfougnier! The PR is updated and ready for another look.

@blackjackkent I added some new checks to site-settings>cloudflare-analytics, so could take an extra close look at that?

**Comment by blackjackkent** [comment]
This looks good to me! @sixhours can you confirm your feedback is all taken care of?

File: client/my-sites/site-settings/cloudflare.js [hunk 11]
```diff
@@ -30,32 +40,76 @@ const Cloudflare = () => {
...
+					{ ! sitePlan && <QuerySitePlans siteId={ siteId } /> }
+					<SettingsSectionHeader title={ translate( 'CDN' ) } />
+					<CompactCard>
 						<div className="site-settings__cloudflare">
+							<div className="site-settings__cloudflare-illustration">
+								<img src={ jetpackIllustration } alt="" />
+							</div>
 							<div className="site-settings__cloudflare-text">
+								<p className="site-settings__cloudflare-title">
+									{ translate( 'Jetpack Site Accelerator' ) }
```
Comment (a8ci18n) [inline client/my-sites/site-settings/cloudflare.js:52]: :information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **30** times:
`translate( 'Site Accelerator' )` *ES Score: 6*

**Comment by sixhours** [comment]
Code changes look good to go! 👍

**Review by blackjackkent — APPROVED** [review]

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/5543333

Thank you @donlair for including a screenshot in the description! This is really helpful for our translators.

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/5543333).

**Pull request merged**